### DISCUSSION TITLE
2.4.20 - Unhealthy Fleet Agents

### DISCUSSION THREAD

**Discussion by sjnhawk** [Security-Onion-Solutions/securityonion#11485] [2.4] [answered]
### Version

2.4.20

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

4

### RAM

32GB

### Storage for /

4TB

### Storage for /nsm

4TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

I'm tracking that other users have had similar issues:
#11148 

I've tried this fix:
`Remove the Elastic Defend integration by navigating to Elastic Fleet -> Agent Policies -> endpoints-initial -> Actions -> Delete integration. Then run the following on the Manager:

sudo rm -f /opt/so/state/eaintegrations.txt
sudo so-elastic-fleet-integration-policy-load

If you are running a distributed grid with a Manager or ManagerSearch, you may also need to add the following role to Elasticsearch:
'transform'
sudo vi +24 /opt/so/saltstack/default/salt/elasticsearch/config.map.jinja 
so-elasticsearch-restart`

And am now getting this error along with unhealthy agents:
![image](https://github.com/Security-Onion-Solutions/securityonion/assets/67609108/50d387b5-ef37-4b57-99a9-8b1ce94056be)

This is a very simple home network SO setup with only 8 agents.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by sjnhawk** [comment]
I also noticed that after doing the "fix" Osquery manager is no longer configured.

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/67609108/addd962f-022d-4419-ab1f-11f82778e480)

**Comment by sjnhawk** [comment]
Ran the fix again, and now everything is healthy.

With that, any ideas on the Osquery Manager?

**Comment by defensivedepth** [comment]
Was this an upgrade or fresh install of 2.4.20?

**Reply by sjnhawk** [reply]
@defensivedepth this was a fresh install. 

Unfortunately an upgrade from 2.4.10 to 2.4.20 failed on me and had to start fresh. Similar to #11481

**Reply by defensivedepth** [reply]
This should not happen with a fresh install of 2.4.20.  Does setup complete without errors?

**Reply by sjnhawk** [reply]
@defensivedepth Yes, setup went smooth with no errors. Osquery stopped working after applying the fix from #11148

**Reply by sjnhawk** [reply]
*I got the unhealthy agents following the fresh install, tried the fix in #11148, then Osquery manager stopped working. Hope that helps.

**Reply by sjnhawk** [reply]
@defensivedepth any ideas on how to get the Osquery back up?

**Reply by defensivedepth** [reply, marked as answer]
We released a hotfix for this issue yesterday. Please run soup and it will apply the hotfix, and should get it all fixed up for you.
